fedimint-cli/src/lib.rs

```diff
@@ -518,6 +518,7 @@ impl FedimintCli {
         let mut client_builder = Client::builder(db);
         client_builder.with_module_inits(self.module_inits.clone());
         client_builder.with_primary_module(1);
+        client_builder.migrate_database().await.map_err_cli_io()?;
```

Comment:
Probably not the right place to hook it. Seems to me that that `ClientBuilder::make_client_builder` would be better. Why? Because at this stage we are not attempting to build the client yet, so why would it start doing anything.

BTW. I have a recovery PR that refactors bunch of things here so there will be trouble (merge conflicts). But not a big deal.

Reply:
This is called in `make_client_builder` right? Maybe I'm just not understanding your comment.

I put it here because this seemed to be the earliest point at which the database was loaded, and I wanted to minimize the chance that we try to read anything from the database before the db can be migrated (even though we still actually read the config from the db before migration currently).

Alternatively I could move this to be a function on `Client` instead of `ClientBuilder`. Then it would be the external library's responsibility to call `migrate_database` before doing anything with the database. But then we'd need to be careful to essentially never touch the database instead of `ClientBuilder` because it could break.

This is a small addition, so I'm ok with handling the merge conflicts :)

Reply:
> Seems to me that `ClientBuilder::make_client_builder` would be better. 

My mistake. I was to write `ClientBuilder::build_stopped`. This is where the `Client` is actually built, recovery/init functions are called. We should do migrations right before that.

Reply:
Cool, works for me.

Reply:
Could there be any reason not to migrate the DB? In that case we could add a boolean to the builder that controls if migration will run on construction.

Reply:
New code will expect new datastructures, it's possibly not even safe to call anything in the module without migrations up to date.

Reply:
Moved to `build_stopped`